src/datastore/postgres-notifier.ts

```diff
@@ -99,15 +109,30 @@ export class PgNotifier {
   }
 
   public async close() {
-    await this.subscriber.unlisten(this.pgChannelName);
-    await this.subscriber.close();
+    await this.subscriber?.unlisten(this.pgChannelName);
+    await this.subscriber?.close();
   }
 
   private async notify(notification: PgNotification) {
-    await this.subscriber
-      .notify(this.pgChannelName, { notification: notification })
-      .catch(error =>
-        logError(`Error sending PgNotifier notification of type: ${notification.type}`, error)
-      );
+    if (this.subscriber) {
+      await this.subscriber
+        .notify(this.pgChannelName, { notification: notification })
+        .catch(async error => {
+          if (
+            error instanceof Error &&
+            error.message === 'Client has encountered a connection error and is not queryable'
```

Comment:
@zone117x this is horrible lol but unfortunately `pg` doesn't throw anything useful for me to catch here other than an `Error` with that specific string... suggestions are welcome for how I can improve this

Reply:
Ah, we have existing similarly horrible code here due to the pg lib not throwing very good Error types: https://github.com/hirosystems/stacks-blockchain-api/blob/a0fd2fe0baf396866f5ca8bb8b9ab91807f02349/src/datastore/postgres-store.ts#L673-L688

I'm actually confused now looking at the `pg` lib source. The error strings you've included are from `node_modules/pg/lib/native/client.js`, using `pg-native` which I thought was an opt-in mode that we don't use. 

For reference, the existing pg connection error handling code is checking strings defined in `node_modules/pg/lib/client.js` (non-native version). 

Does the pg-notifier lib opt-in to the native version, or is there some automatic process where sometimes it uses native when possible?

The error string seem different between the new two modes so that's the only concern I have here.

Reply:
I took that string from the error message @CharlieC3 reported:
```
{"level":"error","message":"Error sending PgNotifier notification of type: microblockUpdate Client has encountered a connection error and is not queryable","stack":"Error: Client has encountered a connection error and is not queryable\n    at /app/node_modules/pg/lib/client.js:563:27\n    at processTicksAndRejections (node:internal/process/task_queues:78:11)","timestamp":"2022-01-05T18:43:18.878Z"}
```
I basically went to that exact location in `pg/lib/client.js` to get the string

Reply:
Ah okay, looks good to me then!